### ISSUE TITLE
Question: Would a CMake Package config be useful

### ISSUE THREAD

**Issue by hjmallon** [libjpeg-turbo/libjpeg-turbo#339]
As part of integration downstream I will be implementing a patch to make a simple CMake package (see https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html) install. Would that be something that upstream (here) would find useful?

Changes would be something like in my test branch here https://github.com/hjmallon/libjpeg-turbo/commit/0be1fc7b565ae206552c29acfedfe38eaa6bd6f8

**Comment by dcommander** [comment]
Possibly.  Since we already support pkg-config, which can be used through CMake by doing something something similar to this:

```
set(CMAKE_PREFIX_PATH /opt/libjpeg-turbo)
find_package(PkgConfig REQUIRED)
pkg_search_module(TURBOJPEG REQUIRED libturbojpeg)

link_directories(${TURBOJPEG_LIBDIR})
add_executable(tjexample tjexample.c)
target_include_directories(tjexample PUBLIC ${TURBOJPEG_INCLUDE_DIRS})
target_link_libraries(tjexample ${TURBOJPEG_LIBRARIES})
```

I'm curious what the advantages of a CMake package would be, apart from eliminating the external pkg-config dependency.

**Comment by hjmallon** [comment]
It is basically the same as pkg-config in my experience, but avoids having to use pkg-config on Windows (which I assume from this StackOverflow is a pain https://stackoverflow.com/questions/1710922/how-to-install-pkg-config-in-windows)

**Comment by dcommander** [comment]
Submit a PR against master.  I'll review it.

**Comment by Youw** [comment]
> I'm curious what the advantages of a CMake package would be, apart from eliminating the external pkg-config dependency.

1) It makes the usage in CMake nicer:
```cmake
find_package(turbojpeg REQUIRED)

add_executable(tjexample tjexample.c)
target_link_libraries(tjexample turbojpeg)
```
You write less code/get more result.

2) 
> apart from eliminating the external pkg-config dependency

That is a big deal. Specially if you (like me) a Windows user/developer (too).

3) It makes life easier when you cross-compile a CMake project.
If I have a CMake package, I can do something like:
```sh
cmake .. -DCMAKE_TOOLCHAIN_FILE=<e.g. Android toolchain> -Dturbojpeg_ROOT=<path to installed build for android> ...
```

Using pkg-config, I cannot specify nice and very explicit `-Dturbojpeg_ROOT` parameter (which I find easy to maintain), but instead I have to specify `CMAKE_PREFIX_PATH` which in time becomes a semicolon separated list of whole bunch of 3rd-party libraries which is in best case is unreadable.

**Comment by dcommander** [comment]
Thanks.  This feature will be included in libjpeg-turbo 2.1.  I just haven't had time to integrate it yet.

**Comment by dcommander** [comment]
I am trying to eat my own dog food and use the libjpeg-turbo CMake package config file in a downstream project, but it doesn't appear to work.  `find_package()` finds the official libjpeg-turbo package with no problem as long as I set `CMAKE_PREFIX_PATH` to **/opt/libjpeg-turbo**, and the targets are successfully imported.  However, any attempt to link with the imported targets, even in the same directory scope, fails.  What am I doing wrong?
